README.md

```diff
@@ -1,5 +1,142 @@
 
-Run ```docker run -p 5432:5432 -e POSTGRES_PASSWORD=password -e POSTGRES_USER=user postgres``` before starting 
+    - **Response:**
+        - Status Code: 200 OK
+        - Body: Success message confirming deletion.
+
+4. **Update a customer**
+    - **Request:**
+        - Method: PUT
+        - URL: `/api/v1/customers/{customerId}`
+        - Example Request:
+          ```
+          PUT /api/v1/customers/123
+          Content-Type: application/json
+   
+          {
+            "name": "Updated Name",
+            "email": "updated@example.com",
+            "age": 35
+          }
+          ```
+
+    - **Response:**
+        - Status Code: 201 Created
+        -  Example Response:
+             ```
+             {
+             "id" 123
+             "name": "Updated Name",
+             "email": "john@example.com",
+             "age": 35
+             }
+             ```
+
+ These endpoints are defined in the `CustomerController` class.
+
+### How the Test works
+- The  integration test spins up the controller, service and database layer. As part of the test it uses the TestContainers 
+library to set up a Postgres container.
+- The ``CustomerControllerTest.Initializer`` static class overrides the application properties in ``application.yml`` so that our 
+application points to the Postgres container
+- Next liquibase runs the database migrations defined in the `migrations.xml` file to set up the customer table in the PostgreSQL container so that the container has a Customer table.
+- After, our tests use JDBI to connect to the Postgres container to set up test data into our Postgres container (e.g. adding and removing customers from our table) for our tests to work. 
+An @AfterEach hook runs after each test to clear away the test data.
+- Rest Assured is used in the test to verify the correct behavior of the API endpoints by sending requests and checking responses against expected values.
+
+###  How the Application works
+- The application uses SpringDoc OpenAPI to generate Swagger documentation. 
+The SwaggerUI is configured in the `application.yml` file with path to access it. It can be accessed at http://localhost:9000/swagger-ui.html
+when the application is running.
+- The `application.yml` file contains the app's configuration details, like Liquibase config for database.
+- Liquibase reads and applies database migrations defined in `00001_create_table_customer.sql` and creates the customer table
+- Jackson is used for JSON serialisation and deserialisation, For example it uses NewCustomerRequest to deserialize the body of the POST request and uses List to serialize to JSON. 
+- Hibernate is the JPA implementation and is an ORM which uses `@Entity` to map onto the Customer table in the database
+- The CustomerRepo interface extends JpaRepository<Customer, Integer>, which provides some built-in methods for database operations (e.g., save(), findById() etc) which are available to us without explicitly defining them
```

Comment:
"Hibernate is an ORM that implements JPA and uses `@Entity` to map onto the `Customer` table"